### PR TITLE
Mimetype detection for hidden files

### PR DESCRIPTION
**Pull request by mwalbeck**
For #3183 

Would this work?

### CODE REVIEW

File: lib/private/Files/Type/Detection.php [hunk 1]
```diff
@@ -167,6 +167,12 @@ public function detectPath($path) {
 		$this->loadMappings();
 
 		$fileName = basename($path);
+
+		// remove leading dot on hidden files with a file extension
+		if (strpos($fileName, '.') === 0 && strpos($fileName, '.', 1)) {
```
Comment (icewind1991) [inline lib/private/Files/Type/Detection.php]:
```php
$fileName = ltrim($fileName, '.');
```

should do the trick 😄

**Review by icewind1991 — CHANGES_REQUESTED** [review]
Otherwise I'm ok with this change

File: lib/private/Files/Type/Detection.php (hunk 1 above)
Comment (mwalbeck) [inline lib/private/Files/Type/Detection.php]: Fixed, thanks

**Review by icewind1991 — APPROVED** [review]

**Review by nickvergessen — APPROVED** [review]
Nice, but this only works for new files right? Old files still have the wrong entry in the database?

**Comment by mwalbeck** [comment]
Correct, old files will still have the wrong mimetype in the filecache.

**Comment by icewind1991** [comment]
iic there is a repair command to fix mimetypes, not sure if it's applicable here

**Pull request merged**